### PR TITLE
Smoother app leave and exposure alert transition

### PR DESCRIPTION
**Pull request by AlanSl**
For https://github.com/covid-alert-ny/covid-green-app/issues/575#issuecomment-731450707 - also @floridemai you were right that something was off with the placement of the `!app.loading` in https://github.com/covid-alert-ny/covid-green-app/pull/577

### CODE REVIEW

File: src/components/views/settings/leave.tsx [hunk 1]
```diff
@@ -29,16 +30,24 @@ export const Leave: FC<{navigation: NavigationProp<any>}> = ({navigation}) => {
...
-      await app.clearContext();
-      await forget();
-      await Haptics.notificationAsync(Haptics.NotificationFeedbackType.Success);
-
       PushNotification.setApplicationIconBadgeNumber(0);
+      const deviceLanguage = getDeviceLanguage();
+      const willRestart = i18n.dir(i18n.language) !== i18n.dir(deviceLanguage);
 
-      RNRestarter.Restart();
+      await forget();
```
Comment (floridemai) [inline src/components/views/settings/leave.tsx:37]: Is "restart" not needed anymore?

File: src/components/views/settings/leave.tsx [hunk 2]
```diff
@@ -29,16 +30,24 @@ export const Leave: FC<{navigation: NavigationProp<any>}> = ({navigation}) => {
...
+      await app.clearContext();
+      Haptics.notificationAsync(Haptics.NotificationFeedbackType.Success);
+      await i18n.changeLanguage(getDeviceLanguage());
```
Comment (floridemai) [inline src/components/views/settings/leave.tsx]: Can you use the `deviceLanguage` defined above?

File: src/components/views/settings/leave.tsx [hunk 3]
```diff
@@ -29,16 +30,24 @@ export const Leave: FC<{navigation: NavigationProp<any>}> = ({navigation}) => {
...
+      if (!willRestart) {
```
Comment (floridemai) [inline src/components/views/settings/leave.tsx:43]: if it `willRestart` will it restart?

**Comment by AlanSl** [comment]
@floridemai I've added a comment that should clarify the `willRestart` thing.

The important things for fixing the previous crash are awaiting the exposure changes and ensuring there isn't a navigation and app restart at the same time; the way I did it before added an extra restart that made the process not crash but janky, this keeps it to one restart or navigation.

**Review by floridemai — APPROVED** [review]

**Pull request merged**